spark/src/main/java/com/netflix/iceberg/spark/source/Writer.java

```diff
@@ -235,21 +224,23 @@ public String toString() {
 
         if (useObjectStorage) {
           // try to get db and table portions of the path for context in the object store
-          String context = pathContext(baseDataPath);
+          String context = pathContext(new Path(baseDataPath));
           String objectStore = properties.get(OBJECT_STORE_PATH);
           Preconditions.checkNotNull(objectStore,
               "Cannot use object storage, missing location: " + OBJECT_STORE_PATH);
-          Path objectStorePath = new Path(objectStore);
 
           outputPathFunc = key -> {
             String partitionAndFilename = key.toPath() + "/" + filename;
             int hash = HASH_FUNC.apply(partitionAndFilename);
-            return new Path(objectStorePath,
-                String.format("%08x/%s/%s", hash, context, partitionAndFilename));
+            return UriBuilder.fromUri(URI.create(objectStore))
```

Comment:
The Path/URI methods cause problems with escape characters. This should use strings for reliability.